MetaMask/metamask-extension
# feat: add api spec test infrastructure

## **Description**
This PR adds tests using [api-specs](https://github.com/MetaMask/api-specs) via the [@open-rpc/test-coverage](https://github.com/open-rpc/test-coverage) tool.

Check out the [test report](https://output.circle-artifacts.com/output/job/fed0bf65-b864-4c37-a501-0050498dbc50/artifacts/0/html-report/index.html) on CircleCI as an artifact.


[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/24132?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2304

## **Manual testing steps**

1. run `yarn build:test`
2. run `yarn test:api-specs`
3. see test report

## **Screenshots/Recordings**
[Link to video from PI lane demos](https://www.notion.so/metamask-consensys/31-of-May-2024-04770f89aa924ab4b978aee89ce0af8c?pvs=4#46235d960d0e4da7ba7dcccac74f8606)
![image](https://github.com/MetaMask/metamask-extension/assets/364566/fe41484e-c92a-4a32-9113-3e9c0d02df71)
![image](https://github.com/MetaMask/metamask-extension/assets/364566/23a0d201-e572-4b3e-a7f1-e01d6992c5a1)
![image](https://github.com/MetaMask/metamask-extension/assets/364566/6d293610-d4ab-47d5-9eaf-5854203ad9f8)

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
